apps/docs/content/guides/ai/examples/semantic-image-search-amazon-titan.mdx

```diff
@@ -0,0 +1,228 @@
+            "embeddingConfig": {"outputEmbeddingLength": 1024},
+        }
+    )
+
+
+def get_embedding_from_titan_multimodal(body):
+    response = bedrock_client.invoke_model(
+        body=body,
+        modelId="amazon.titan-embed-image-v1",
+        accept="application/json",
+        contentType="application/json",
+    )
+
+    response_body = json.loads(response.get("body").read())
+    print(response_body)
+    return response_body["embedding"]
+
+
+def encode_image(file_path):
+    base64_string = readFileAsBase64(file_path)
+    body = construct_bedrock_image_body(base64_string)
+    emb = get_embedding_from_titan_multimodal(body)
+    return emb
+```
+
+Next, create a `seed` method, which will create a new Supabase Vector Collection, generate embeddings for your images, and upsert the embeddings into your database:
+
+```python
+def seed():
+    # create vector store client
+    vx = vecs.create_client(DB_CONNECTION)
+
+    # create a collection of vectors with 512 dimensions
+    try:
+        images = vx.create_collection(name="image_vectors", dimension=1024)
+    except:
+        images = vx.get_collection(name="image_vectors")
+
+    # Generate image embeddings with Amazon Titan Model
+    img_emb1 = encode_image('./images/one.jpg')
+    img_emb2 = encode_image('./images/two.jpg')
+    img_emb3 = encode_image('./images/three.jpg')
+    img_emb4 = encode_image('./images/four.jpg')
+
+    # add records to the *images* collection
+    images.upsert(
+        records=[
+            (
+                "one.jpg",       # the vector's identifier
+                img_emb1,        # the vector. list or np.array
+                {"type": "jpg"}  # associated  metadata
+            ), (
+                "two.jpg",
+                img_emb2,
+                {"type": "jpg"}
+            ), (
+                "three.jpg",
+                img_emb3,
+                {"type": "jpg"}
+            ), (
+                "four.jpg",
+                img_emb4,
+                {"type": "jpg"}
+            )
+        ]
+    )
+    print("Inserted images")
+
+    # index the collection for fast search performance
+    images.create_index()
+    print("Created index")
+```
+
+Add this method as a script in your `pyproject.toml` file:
+
+```toml
+[tool.poetry.scripts]
+seed = "image_search.main:seed"
+search = "image_search.main:search"
+```
+
+After activating the virtual environtment with `poetry shell` you can now run your seed script via `poetry run seed`. You can inspect the generated embeddings in your Supabase Dashboard by visiting the [Table Editor](https://supabase.com/dashboard/project/_/editor), selecting the `vecs` schema, and the `image_vectors` table.
+
+## Perform an image search from a text query
+
+With Supabase Vector we can easily query our embeddings. We can use either an image as search input or alternative we can generate an embedding from a string input and use that as the query input:
+
+```python
+def search(args=sys.argv):
```

Comment:
Nit: this looks a little funky

There's nothing wrong with it, but it's coupled tightly to command line args and it isn't obvious how to override for non CLI use.

If you wanted to call this from code you'd have to write

```python
search(args = ["some query term"])
```

alternatively if the signature was 
```python
def search(query_term: Optional[str] = None):
   if query_term is None:
      query_term = sys.argv[0]
   ...
```
then its clearer from code and still usable via CLI